### PR TITLE
Enhance/133/restore linux support

### LINKED ISSUE #133: Restore Linux Support
As part of the migration to the Fusion sync service in Tokaido 1.6, we removed Linux support. As luck would have it, we need to re-add this service in order to run Tokaido in CircleCI (as part of the updated `tok test` functionality planned for Tokaido 1.9. 

Acceptance Criteria: 
- [x] Re-add Linux support in the Tokaido CLI
- [x] Re-add the Unison sync service and systemctl support for Linux users 
- [x] Ensure that Linux users use the Unison sync service, while MacOS users can choose between Unison, Fusion, and Docker
- [x] Ensure that there is a documentation page explaining the sync strategies, advantages and disadvantages of each, and which one we recommend
- [ ] Verify that Tokaido can now run inside CircleCI when using the `machine` executor.

### PR DESCRIPTION
**Pull request by otakumike**
fixes #133 

I've tested this pretty extensively on macOS and Linux, both seem to be working well. 

On Linux, Unison is the default and only support sync service. On macOS, Unison and Docker are officially supported and anyone still using Fusion gets a warning message that Fusion will be removed in the next release.

### CODE REVIEW

File: conf/get_config.go [hunk 1]
```diff
@@ -43,6 +50,18 @@ func GetProjectPath() (path string) {
...
+// GetGlobalProjectSettings returns the current global conf object for the current project
+func GetGlobalProjectSettings() (*Project, error) {
+	c := GetConfig()
+	for _, v := range c.Global.Projects {
+		if v.Name == c.Tokaido.Project.Name {
+			return &v, nil
+		}
+	}
+
+	return &Project{}, fmt.Errorf("unable to find global project configuration")
```
Comment (yjimk) [inline conf/get_config.go]: Nitpick, maybe ok in context anway. unable -> Unable?

File: conf/set_config.go [hunk 2]
```diff
@@ -57,6 +59,64 @@ func SetConfigValueByArgs(args []string, configType string) {
 	compareFiles(newConfig, configPath)
 }
 
+// SetGlobalConfigValueByArgs enables the update of global config values by a string slice
+// When working project-level global config, it defaults to updating the active project
+func SetGlobalConfigValueByArgs(args []string) (err error) {
```
Comment (yjimk) [inline conf/set_config.go]: Ah yeah, this is good

File: initialize/main.go [hunk 3]
```diff
@@ -56,7 +57,16 @@ func readProjectConfig(command string) {
 	utils.DebugString("reading project config")
 	pr := fs.ProjectRoot()
 
-	viper.SetDefault("Global.Syncservice", "docker")
+	switch system.CheckOS() {
+	case "osx":
+		viper.SetDefault("Global.Syncservice", "docker")
+
```
Comment (yjimk) [inline initialize/main.go]: Soon: `case: "windows"`

File: conf/set_config.go [hunk 4]
```diff
@@ -186,6 +246,22 @@ func WriteGlobalConfig(ng Global) {
...
+func WriteGlobalProjectSettings(p *Project) error {
+	g := GetConfig().Global
+
+	for k, v := range g.Projects {
+		if v.Name == p.Name {
+			g.Projects[k] = *p
+		}
+	}
+
+	WriteGlobalConfig(g)
```
Comment (yjimk) [inline conf/set_config.go]: No check on `err` here? If this function can never return an error, probably better to remove the return type

File: services/docker/generate.go [hunk 5]
```diff
@@ -94,27 +95,29 @@ func MarshalledDefaults() []byte {
 // UnmarshalledDefaults ...
 func UnmarshalledDefaults() conf.ComposeDotTok {
 	tokStruct := conf.ComposeDotTok{}
-	xdebugImageVersion := conf.GetConfig().Tokaido.Stability
 	phpVersion := conf.GetConfig().Tokaido.Phpversion
+	unisonVersion := version.GetUnisonVersion()
```
Comment (yjimk) [inline services/docker/generate.go]: Could this version check be moved into the switch case for unison?

File: conf/get_config.go (hunk 1 above)
Comment (otakumike) [reply conf/get_config.go]: "best practice" ... 🙄 

https://github.com/golang/go/wiki/CodeReviewComments#error-strings

File: conf/set_config.go (hunk 4 above)
Comment (otakumike) [reply conf/set_config.go]: good find - will fix !

File: services/docker/generate.go (hunk 5 above)
Comment (otakumike) [reply services/docker/generate.go]: also a good point - fixed thanks

**Pull request merged**